tests/publiccloud/register_system.pm

```diff
@@ -23,21 +23,26 @@ use publiccloud::utils "select_host_console";
 sub run {
     my ($self, $args) = @_;
 
-    my @addons = split(/,/, get_var('SCC_ADDONS', ''));
-
-    select_host_console();    # select console on the host, not the PC instance
-
-    # note: ssh_script_retry dies on failure
-    $args->{my_instance}->retry_ssh_command("sudo SUSEConnect -r " . get_required_var('SCC_REGCODE'), timeout => 420, retry => 3) if (get_var('FLAVOR') =~ 'BYOS');
-
-    for my $addon (@addons) {
-        if (is_sle('<15') && $addon =~ /tcm|wsm|contm|asmm|pcm/) {
-            ssh_add_suseconnect_product($args->{my_instance}->public_ip, get_addon_fullname($addon), '`echo ${VERSION} | cut -d- -f1`') unless ($addon eq '');
-        } elsif (is_sle('<15') && $addon =~ /sdk|we/) {
-            ssh_add_suseconnect_product($args->{my_instance}->public_ip, get_addon_fullname($addon), '${VERSION_ID}') unless ($addon eq '');
-        } else {
-            ssh_add_suseconnect_product($args->{my_instance}->public_ip, get_addon_fullname($addon)) unless ($addon eq '');
+    if (get_var('FLAVOR') =~ 'BYOS') {
```

Comment:
There might be multiple flavours - not just **BYOS** - but as we schedule only BYOS and on-demand, I'd keep this as it is.

Reply:
I see a similar problem there: As of now we are by default excluding the registration and only for certain flavour including it (`BYOS`). It should be the opposite, but it's not easily done with the current scheme.
So that's not the best solution, but it does the job for now.